### ISSUE TITLE
XRechnung 2.0

### ISSUE THREAD

**Issue by TomcatBB** [stephanstapel/ZUGFeRD-csharp#93]
Hello,

is support for XRechnung 2.0 planned and when can we expect it?
Because from 01.01.2021 on only XRechnung 2.0 are valid and the XRechnung 1.2.2 become deprecated.

Best Regards

**Comment by stephanstapel** [comment]
Hi,

I'm so far sticking to the standards produced by https://www.ferd-net.de/.
Where did you get the information that XRechnung 1.2.2 is not valid from 2021/01/01 ?

**Comment by TomcatBB** [comment]
Hey, 

its stated on the German KOSIT Page that are responsible for the XRechnung standard.
https://www.xoev.de/xrechnung-16828

And also on the Mustang Project ( The Java Implementation of Zugferd and XRechnung ) 
https://www.mustangproject.org/xrechnung/?lang=de

Best Regards

**Comment by stephanstapel** [comment]
Interesting. Looks like the deprecation is concerning sending invoices to German government (at least this is stated on the mustang site).

The information bundle comes with a version history:

![image](https://user-images.githubusercontent.com/2912080/95619972-bdfc2c80-0a6f-11eb-8ec4-e66f31f048a0.png)

I guess I would not implemented myself:
https://projekte.kosit.org/xrechnung/xrechnung/-/issues/60

This is one is quite simple:
https://projekte.kosit.org/xrechnung/xrechnung/-/issues/59
and depends on the implementation here:  https://github.com/stephanstapel/ZUGFeRD-csharp/issues/92

I don't think these are related to the library:
https://projekte.kosit.org/xrechnung/xrechnung/-/issues/61
https://projekte.kosit.org/xrechnung/xrechnung/-/issues/62
https://projekte.kosit.org/xrechnung/xrechnung/-/issues/63
but are more things about documentation and usage of the standard

This one is already implemented:
https://projekte.kosit.org/xrechnung/xrechnung/-/issues/65

what's your opinion?

**Comment by TomcatBB** [comment]
Yes you are right, as i understand the most parts are not touching the concrete implementation except for this one:
https://projekte.kosit.org/xrechnung/xrechnung/-/issues/60

Maybe someone should include this also to verify that all features are available. As i know for now, we dont need this feature in our project.

But what is important that the Version Identifier in the XML has to change from 
`<ram:ID>urn:cen.eu:en16931:2017#compliant#urn:xoev-de:kosit:standard:xrechnung_1.2</ram:ID>`
to
`<ram:ID>urn:cen.eu:en16931:2017#compliant#urn:xoev-de:kosit:standard:xrechnung_2.0</ram:ID>`

So maybe we need a new Profile called XRechnung20 for instance to separate this 2 or what do you mean?

**Comment by stephanstapel** [comment]
Thanks for getting back. As written above, I'd not implement sub items as this is a very rare case (and honestly, one thing is to write invoices with these lines, the other thing is to process them in the backend/ ERP systems).

Concerning the versioning I'm not sure yet how to progress. XRechnung20 is no official profile for ZUGFeRD/ Factur-X, so I'd not implement it like this. Maybe it's rather a new version. I'd definitely wait until the end before moving on.

**Comment by TomcatBB** [comment]
Ok got it :-) As i sayed we also dont need this special feature.
So if someone needs this, he or she can do a pull request, right?

For the Version its difficult, yes as i understand right now its not allowed to already use XRechnung 2.0 but starting from 1.1.2021 its mandatory and the XRechnung 1.2.2 is not longer allowed. So i wonder how to do the transform, or to be prepared for the transform, without having the need of updating the NuGet Reference in the last days of the year, and the need for deploying a new Version of our Software in the last days of the year. You know what i mean?

Maybe a solution could be a switch in the XML Generation process that writes the XRechnung Version depenting on the current Date?

**Comment by stephanstapel** [comment]
Yes, that's right. I'd be happy to review and integrate a pull request.

Concerning ZUGFeRD: In the meantime I learned that there won't be a new version containing XRechnung 2.0 reference profile. So I might need to find a solution.

I created specific tickets:
https://github.com/stephanstapel/ZUGFeRD-csharp/issues/94
https://github.com/stephanstapel/ZUGFeRD-csharp/issues/95

and close the ticket here - hope this is alright for you.
